LCD overflow bug and audio demo bug fixes

JColvin91 opened the pull request:
Changed LCD.c's to limit the number of characters shown on the display from 40 to 32. This was creating a bug in the OOB where the LCD would display the incorrect combinations of LCD string values. 
40 characters can be used with the KS0066 display controller but is only supported when 2 displays are used in series with one another, which I doubt will be a primary use case for the Basys MX3.
Also added a line in the audio_Close function in the audio.c to allow for users to disable and re-enable the same mode repeatedly without being forced to switch audio modes.

JColvin91 committed:
Update lcd.c

fixed overflow runtime error that is only compatible with multiple LCD screens

JColvin91 committed:
Update lcd.c

fixed overflow error

JColvin91 committed:
Update lcd.c

fix error

JColvin91 committed:
Update lcd.c

JColvin91 committed:
Update lcd.c

JColvin91 committed:
Update lcd.c

JColvin91 committed:
Update lcd.c

JColvin91 committed:
Update lcd.c

JColvin91 committed:
Update lcd.c

JColvin91 committed:
Update lcd.c

JColvin91 committed:
Update lcd.c

JColvin91 committed:
Update audio.c

Add in reset for bAudioParameter so users can flip the same switch on and off for the audio demo and have it continue to work.

JColvin91 committed:
Update audio.c

minor change to allow for an audio mode to be enabled, disabled, and then re-enabled again.

JColvin91 committed:
Update audio.c

JColvin91 committed:
Merge pull request #1 from JColvin91/BasysMX3-Audio-reset-fix

Basys mx3 audio reset fix

sbobrowicz commented:
It looks like something got messed up, there are way more commits than there should be. We definitely don't want to merge this in its current state or it will mess up the log. I'd recommend cloning and using the github client rather than using the browser based text editor to make changes. Also please choose more unique and descriptive commit titles. 

I'll let Cristian or the others who are more familiar with the code comment on your changes.

JColvin91 commented:
Changed to the cloning method with the desktop client and redid the commit so it's a more reasonable number of changes (15 vs 55+). Also changed the commit title and made sure the description adequately explains the changes.

Let me know if you need any other changes to be made before the proposed changes can be reviewed.
